src/include/daos_obj.h

```diff
@@ -105,6 +105,32 @@ enum {
 	DAOS_OF_MASK		= ((1 << OID_FMT_FEAT_BITS) - 1),
 };
 
+/** Number of bits reserved in IO flags bitmap for conditional checks.  */
+#define IO_FLAGS_COND_BITS	8
+
+enum {
+	/** Flag on Object */
+	DAOS_OP_OBJ		= (1 << 0),
```

Comment:
Are there scenarios where we would want something like a conditional update on the akey but a conditional insert on the dkey?   There doesn't seem to be any way to specify such if you specify the flags in this way

Reply:
conditional insert on dkey, means fail if dkey exists. so the akey check doesn't make sense.

i guess conditional update on dkey and conditional insert on akey, might make sense, although not sure if there is a real use case for it. maybe we have to go with my other proposal (from mailing list)